R/deprecated.R

```diff
@@ -1,7 +1,55 @@
+
+
+rlang_inform <- function(..., msg = paste0(...), frequency = "regular", id = msg) {
+  rlang::inform(
+    message = msg,
+    .frequency = frequency,
+    .frequency_id = id
+  )
+}
+rlang_warn <- function(..., msg = paste0(...), frequency = "regular", id = msg) {
+  rlang::warn(
+    message = msg,
+    .frequency = frequency,
+    .frequency_id = id
+  )
+}
+
 #' Print message for deprecated functions in Shiny
 #'
 #' To disable these messages, use `options(shiny.deprecation.messages=FALSE)`.
 #'
+#' @inheritParams lifecycle::deprecate_soft
+#' @keywords internal
+#' @describeIn soft-deprecated Passes arguments to [lifecycle::deprecate_soft()]
+shinySoftDeprecated <- function(
+  when, what, with = NULL, details = NULL, id = NULL,
+  env = rlang::caller_env(1)
+) {
+  if (is_false(getOption("shiny.deprecation.messages"))) {
+    return(invisible())
+  }
+
+  lifecycle::deprecate_soft(when, what, with = with, details = details, id = id, env = env)
```

Comment:
Might need to change this. From the docs:

> deprecate_soft() warns only if the deprecated function is called from the global environment

Reply:
On second thought, I think we should continue to emit messages instead of warnings. So `rlang_inform`?